module/os/linux/zfs/zfs_uio.c

```diff
@@ -85,9 +84,6 @@ zfs_uiomove_iov(void *p, size_t n, zfs_uio_rw_t rw, zfs_uio_t *uio)
 					    __copy_from_user_inatomic(p,
 					    (iov->iov_base + skip), cnt);
 					pagefault_enable();
-#else
-					return (EFAULT);
-#endif
```

Comment:
It occurs to me do we have one other alternative here worth considering.  The reason `__copy_from_user_inatomic()` is used here is to avoid a very unlikely deadlock scenario, for an even less common workload (described in #7939), which was only ever reported for the 3.10.x kernel (CentOS 7).  The relevant code has changed significantly is newer kernels, and it might not even be possible anymore.  Which is all a long way of saying we could take the `!uio->uio_fault_disable` path here when `!HAVE___COPY_FROM_USER_INATOMIC` and it wouldn't surprise me if we never encounter the original deadlock.

Reply:
I think we could go all in and completely delete that branch to have consistent behavior across architectures.

Reply:
In that case I'd suggest going back to autoconf fast fail. The other approaches seem to create even more ways to fail due to their sophistication.

Reply:
That works for me.